### ISSUE TITLE
ELECTRA: keep track of additional metrics in wandb

### ISSUE THREAD

**Issue by mattivi** [ThilinaRajapakse/simpletransformers#1173]
**Is your feature request related to a problem? Please describe.**
Context: ELECTRA pre-training, e.g. as described in your [article here](https://towardsdatascience.com/understanding-electra-and-training-an-electra-language-model-3d33e3a9660d)
Feature: Im wandb I would like to keep track the following additional metrics, for both train and eval: 
- generator loss, discriminator loss
- mlm accuracy for generator
- auc, precision, recall, f1, mcc, for 'is fake token' in the discriminator 

**Describe the solution you'd like**
Beside the current metrics (lr, global_step, training loss, train_loss, eval_loss, perplexity), is there a quick solution to track the above additional metrics in wandb? I guess modifying language_modeling_model.py and appending into training_progress_scores? 

Thanks a lot

**Comment by mattivi** [comment]
I have already modified language_modeling_model.py to keep track of d_loss and g_loss for wandb.log() and for training_progress_scores dict, actually it is straight forward.
To compute auc, precision, recall etc, I see there are prediction scores in outputs[3] and labels in outputs[4], e.g. from [line 770](https://github.com/ThilinaRajapakse/simpletransformers/blob/a44487d36ce56e329b0e03ac9dc369c42233bcee/simpletransformers/language_modeling/language_modeling_model.py#L770). Is there any reusable code so that it can eventually be used for other models as well in simpletransformers?

Discriminator auc is needed to verify that the discriminator is able to distinguish if inputs are replaced or not (i.e. in case of larger generator the discriminator might not be able to learn). Thanks!
